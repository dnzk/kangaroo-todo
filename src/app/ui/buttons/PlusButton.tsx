import { DetailedHTMLProps, ButtonHTMLAttributes } from "react"
import IconPlus from "@public/plus.svg"
import Image from "next/image"
import style from "./buttons.module.scss"

type Size = "large" | "medium"

type IconButtonProps = DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement> & { size?: Size }

export default function PlusButton(props: IconButtonProps) {
  const imageSize = props.size === "large" ? {} : { width: 30 }

  return (
    <button {...props} className={style.icon}>
      <Image
        src={IconPlus}
        alt="Plus button"
        aria-label="Plus button"
        {...imageSize}
      />
    </button>
  )
}
