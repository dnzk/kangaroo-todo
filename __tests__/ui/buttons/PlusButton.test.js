import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import PlusButton from '../../../src/app/ui/buttons/PlusButton'

describe('PlusButton', () => {
  it('renders correctly', () => {
    const { container } = render(<PlusButton />)
    expect(container).toMatchSnapshot()
  })
})
