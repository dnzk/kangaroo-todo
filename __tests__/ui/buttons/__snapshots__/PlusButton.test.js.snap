// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusButton renders correctly 1`] = `
<div>
  <button
    class="icon"
  >
    <img
      alt="Plus button"
      aria-label="Plus button"
      data-nimg="1"
      decoding="async"
      height="30"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
      style="color: transparent;"
      width="30"
    />
  </button>
</div>
`;
